include/wlr/types/wlr_foreign_toplevel_management_v1.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+#define WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_output.h>
+
+struct wlr_foreign_toplevel_manager_v1 {
+	struct wl_global *global;
+	struct wl_list resources;
+	struct wl_list toplevels; // wlr_foreign_toplevel_handle_v1::link
+
+	struct wl_listener display_destroy;
+
+	struct {
+		struct wl_signal destroyed;
+	} events;
+
+	void *data;
+};
+
+enum wlr_foreign_toplevel_handle_state {
```

Comment:
We decided it's better to use directly enums generated by wayland-scanner.

Reply:
but here the idea is different, the compositor can set a bit mask of states (as opposed to the protocol, where we use an array), then it gets stored into state. If you're against this, I could change it to getting the ready `wl_array` from the compositor.

Reply:
I see. It's fine